src/cloudai/workloads/deepep_benchmark/slurm_command_gen_strategy.py

```diff
@@ -0,0 +1,149 @@
+
+    def _append_sbatch_directives(self, batch_script_content: List[str]) -> None:
+        """
+        Append SBATCH directives and head node detection setup for DeepEP.
+
+        Args:
+            batch_script_content: The list of script lines to append to.
+        """
+        super()._append_sbatch_directives(batch_script_content)
+        self._append_head_node_detection(batch_script_content)
+
+    def _container_mounts(self) -> List[str]:
+        """Return container mounts specific to DeepEP benchmark."""
+        tdef: DeepEPBenchmarkTestDefinition = cast(DeepEPBenchmarkTestDefinition, self.test_run.test)
+        cmd_args: DeepEPBenchmarkCmdArgs = tdef.cmd_args
+
+        # Create config file
+        config_file_path = self.test_run.output_path / "config.yaml"
+        self._generate_config_yaml(config_file_path, cmd_args)
+
+        # Mount config file and results directory
+        mounts = [
+            f"{config_file_path.parent.absolute()}:{config_file_path.parent.absolute()}",
+            f"{self.test_run.output_path.absolute()}:{cmd_args.results_dir}",
+        ]
+
+        return mounts
+
+    def image_path(self) -> str | None:
+        """Return the Docker image path for DeepEP benchmark."""
+        tdef: DeepEPBenchmarkTestDefinition = cast(DeepEPBenchmarkTestDefinition, self.test_run.test)
+        return str(tdef.docker_image.installed_path)
+
+    def generate_test_command(self) -> List[str]:
+        """Generate the test command for DeepEP benchmark."""
+        tdef: DeepEPBenchmarkTestDefinition = cast(DeepEPBenchmarkTestDefinition, self.test_run.test)
+        cmd_args: DeepEPBenchmarkCmdArgs = tdef.cmd_args
+
+        # Determine which benchmark script to run based on mode
+        if cmd_args.mode == "standard":
+            benchmark_script = "/workspace/dp-benchmark/benchmark/benchmark.py"
+        else:  # low_latency
+            benchmark_script = "/workspace/dp-benchmark/benchmark/benchmark_ll.py"
+
+        # Get number of nodes
+        _, nodes = self.system.get_nodes_by_spec(self.test_run.nnodes, self.test_run.nodes)
+        num_nodes = len(nodes) if nodes else self.test_run.nnodes
+
+        # Build torchrun command
+        config_file_path = self.test_run.output_path / "config.yaml"
+        command_parts = [
+            "torchrun",
+            f"--nnodes={num_nodes}",
+            "--nproc_per_node=1",
+            "--rdzv_id=$RANDOM",
+            "--rdzv_backend=c10d",
+            "--rdzv_endpoint=$head_node_ip:29500",
+            benchmark_script,
+            str(config_file_path.absolute()),
+        ]
+
+        return command_parts
+
+    def _generate_config_yaml(self, config_path: Path, cmd_args: DeepEPBenchmarkCmdArgs) -> None:
+        """
+        Generate YAML configuration file for DeepEP benchmark.
+
+        Args:
+            config_path: Path where to write the config file.
+            cmd_args: Command arguments for the benchmark.
+        """
+        config_lines = [
```

Comment:
Is it fair to assume config generation can be generalized into:
```py
for k, v in tdef.cmd_args_dict.items():
    config_lines.append(f"{k}: {str(v).lower()}")
```
?

1. Note, that `cmd_args_dict` is a property of test definition class and there you can control its behavior.
2. `CmdArgs` can contain _any_ fields defined by users, so if a new option is added into your benchmark, you won't need any changes in CloudAI code, only config(s) should be updated.